File: MinIO.py
import os
import io
import urllib3
from minio import Minio
import pandas as pd
import matplotlib.pyplot as plt

# Disable SSL verification warnings
urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)

# Setting the environment variables
os.environ["AWS_ACCESS_KEY_ID"] = "health"
os.environ["AWS_SECRET_ACCESS_KEY"] = "NOentry#23"
os.environ["AWS_SSL_ENABLED"] = "TRUE"
os.environ["AWS_S3_ENDPOINT"] = "127.0.0.1:9000"

# Initialize Minio client
client = Minio(
    endpoint=os.environ["AWS_S3_ENDPOINT"],
    access_key=os.environ["AWS_ACCESS_KEY_ID"],
    secret_key=os.environ["AWS_SECRET_ACCESS_KEY"],
    secure=True,
    http_client=urllib3.PoolManager(cert_reqs='CERT_NONE')
)

# Get files from the MinIO server
bucket_name = 'templategenerator'

# Read the CSV file from MinIO
csv_object = client.get_object(bucket_name, 'options.csv')
csv_data = pd.read_csv(io.BytesIO(csv_object.read()), quotechar="'")
print(csv_data.head())

# Read the image file from MinIO
image_object = client.get_object(bucket_name, 'Ben.jpg')
image_data = plt.imread(io.BytesIO(image_object.read()), format='jpg')
plt.imshow(image_data)
plt.axis('off')
plt.show()

File: MinIO.R
# MinIO R implementation
# https://blog.min.io/minio-apache-arrow-r/
library(devtools)
library(httr)
install_github("nagdevAmruthnath/minio.s3")
library("minio.s3")
library(aws.s3)
library(tidyverse)
library(dplyr)
library(readr)
library(png)
library(jpeg)

# Setting the environment variables
# https://localhost:9001/browser/templategenerator
set_config(config(ssl_verifypeer = 0L, ssl_verifyhost = 0L))
Sys.setenv("AWS_ACCESS_KEY_ID" = "health",
           "AWS_SECRET_ACCESS_KEY" = "NOentry#23",
           "AWS_SSL_ENABLED" = "TRUE",
           "AWS_S3_ENDPOINT" = "127.0.0.1:9000")

# Get files from the MinIO server
get_bucket(bucket = "templategenerator", region = "", use_https = TRUE)
b <- get_bucket(bucket = 'templategenerator', region = "", use_https = TRUE)
csv_file <- aws.s3::s3read_using(FUN = read.csv, object = "options.csv", bucket = b, opts = list(use_https = TRUE, region = ""))
head(csv_file)

image_file <- aws.s3::s3read_using(FUN = readJPEG, object = "Ben.jpg", bucket = b, opts = list(use_https = TRUE, region = ""))
grid::grid.raster(image_file)
